ext/extconf.rb

```diff
@@ -7,8 +7,55 @@
   $CPPFLAGS += $CXXFLAGS
 end
 
+brew = find_executable('brew')
+
+# Automatically use homebrew to discover thrift package if it is
+# available and not disabled via --disable-homebrew.
+use_homebrew = enable_config('homebrew', brew)
+
+if use_homebrew
+  homebrew_package = with_config('homebrew-thrift-package', 'thrift@0.9')
+  $stderr.puts "using Homebrew to locate thrift package '#{homebrew_package}'"
+
+  thrift_prefix = `#{brew} --prefix #{homebrew_package}`.strip
+
+  unless File.exists?(thrift_prefix)
+    $stderr.puts "#{thrift_prefix} does not exist"
+    $stderr.puts "To resolve, `brew install #{homebrew_package}` or pass"
+    $stderr.puts '--with-homebrew-thrift-package=<package> to this build to specify'
+    $stderr.puts 'an alternative thrift package to build against. e.g.:'
+    $stderr.puts
+    $stderr.puts '  $ bundle config --global build.sparsam --with-homebrew-thrift-package=mythrift'
+
+    raise "Homebrew package #{homebrew_package} not installed"
+  end
+
+  $stderr.puts "using Homebrew thrift at #{thrift_prefix}"
+  $CFLAGS << " -I#{thrift_prefix}/include"
+  $CXXFLAGS << " -I#{thrift_prefix}/include"
+  $CPPFLAGS << " -I#{thrift_prefix}/include"
+  $LDFLAGS << " -L#{thrift_prefix}/lib"
+
+  # Also add boost to the includes search path if it is installed.
+  boost_package = with_config('homebrew-boost-package', 'boost')
+
+  boost_prefix = `#{brew} --prefix #{boost_package}`.strip
+
+  if File.exists?(boost_prefix)
+    boost_include = "#{boost_prefix}/include"
+    $stderr.puts("adding Homebrew boost #{boost_include} to include path")
+    $CFLAGS << " -I#{boost_include}"
+    $CXXFLAGS << " -I#{boost_include}"
+    $CPPFLAGS << " -I#{boost_include}"
+  end
+end
+
 have_func("strlcpy", "string.h")
-have_library("thrift")
+
+unless have_library("thrift")
```

Comment:
do we _also_ need to fail if `have_library(boost)` returns false? i assume that compilation would also fail without boost but haven't tried.

Reply:
We don't actually need the boost library since all our functionality appears to be defined in header files (most of boost is headers only).

That being said, I did try to add `have_header('boost/make_shared.hpp')` and validate the presence of boost. But for some reason it didn't work. I spent 5-10 minutes trying to debug before declaring defeat.

Reply:
Apparently `have_header` doesn't work with C++. Good times. https://bugs.ruby-lang.org/issues/4924

Reply:
Nice (and disappointing) find.